### PR TITLE
Maybe wait before purge_queue to avoid doing it too often

### PR DESCRIPTION
**Pull request by netsettler**
There's supposed to be a 60 second gap between purges, so this accommodates that.

This leaves open the question of whether multiple independent agencies might try to do it at the same time, but the normal case where this will be common is in testing, where everything is in one process, so this should usually work, and it may improve some test behaviors.

### CODE REVIEW

File: poetry.lock [hunk 1]
```diff
@@ -2103,10 +2106,12 @@ python-jose = [
 python-magic = [
     {file = "python-magic-0.4.15.tar.gz", hash = "sha256:f3765c0f582d2dfc72c15f3b5a82aecfae9498bd29ca840d72f37d7bd38bfcd5"},
     {file = "python_magic-0.4.15-py2.py3-none-any.whl", hash = "sha256:f2674dcfad52ae6c49d4803fa027809540b130db1dec928cfbb9240316831375"},
+    {file = "python-magic-0.4.18.tar.gz", hash = "sha256:b757db2a5289ea3f1ced9e60f072965243ea43a2221430048fd8cacab17be0ce"},
```
Comment (willronchetti) [inline poetry.lock:2109]: This appears to add a breaking change of some kind based on the build: https://travis-ci.org/github/4dn-dcic/snovault/builds/684210509

File: snovault/elasticsearch/indexer_queue.py [hunk 2]
```diff
@@ -374,16 +384,41 @@ def initialize(self, dlq=False):
...
         """
-        Clear out the queue and dlq completely. You can no longer retrieve
-        these messages. Takes up to 60 seconds.
+        Clear out the queue and dlq completely. You can no longer retrieve these messages.
+        AWS says this operation takes up to 60 seconds, that operations queued before will start to disappear,
+        and that operations queued within 60 seconds after may also disappear.
         """
         for queue_url in [self.queue_url, self.second_queue_url, self.dlq_url]:
             try:
+                self._wait_until_purge_queue_allowed()
```
Comment (willronchetti) [inline snovault/elasticsearch/indexer_queue.py:410]: I'm not sure how I feel about this, I think we should deprecate this method altogether in favor of `clear_queue`, though this sort of protection is probably good for when we do use it.

Comment (netsettler) [reply snovault/elasticsearch/indexer_queue.py:410]: Yeah, I don't like this operation and agree with you for as far as this goes. I think this operation must be something they don't intend you to use directly. I have some further thoughts about this I'll talk to you about in another forum that's more interactive.

File: poetry.lock (hunk 1 above)
Comment (netsettler) [reply poetry.lock:2109]: A patch change that's a breaking change? Sigh. OK.

Comment (willronchetti) [reply poetry.lock]: Yeah, super irritating for sure but as you saw seemed to in fact be the problem.

**Pull request merged**